test/e2e/e2e-ttl/e2e-ttl-es/src/test/java/org/apache/skywalking/e2e/StorageTTLITCase.java

```diff
@@ -0,0 +1,278 @@
+
+        ensureSendingMetricsWorks(
+            builder,
+            startTime.toEpochSecond(ZoneOffset.UTC) * 1000,
+            endTime.toEpochSecond(ZoneOffset.UTC) * 1000,
+            queryStart,
+            queryEnd,
+            "DAY"
+        );
+
+        shouldBeEmptyBetweenTimeRange(queryStart, queryEnd, "DAY");
+    }
+
+    @Test(timeout = 360000)
+    public void monthMetricsDataShouldBeRemovedAfterTTL() throws Exception {
+
+        final ServiceMeshMetric.Builder builder = ServiceMeshMetric
+            .newBuilder()
+            .setSourceServiceName("e2e-test-source-service")
+            .setSourceServiceInstance("e2e-test-source-service-instance")
+            .setDestServiceName("e2e-test-dest-service")
+            .setDestServiceInstance("e2e-test-dest-service-instance")
+            .setEndpoint("e2e/test")
+            .setLatency(2000)
+            .setResponseCode(200)
+            .setStatus(SUCCESS)
+            .setProtocol(Protocol.HTTP)
+            .setDetectPoint(DetectPoint.server);
+
+        final LocalDateTime now = LocalDateTime.now();
+        final LocalDateTime startTime = now.minusMonths(SW_STORAGE_ES_MONTH_METRIC_DATA_TTL + 1);
+        final LocalDateTime endTime = startTime.plusMinutes(1);
+
+        final LocalDateTime queryStart = startTime;
+        final LocalDateTime queryEnd = now.minusMonths(SW_STORAGE_ES_MONTH_METRIC_DATA_TTL);
+
+        ensureSendingMetricsWorks(
+            builder,
+            startTime.toEpochSecond(ZoneOffset.UTC) * 1000,
+            endTime.toEpochSecond(ZoneOffset.UTC) * 1000,
+            queryStart,
+            queryEnd,
+            "MONTH"
+        );
+
+        shouldBeEmptyBetweenTimeRange(queryStart, queryEnd, "MONTH");
+    }
+
+    private void shouldBeEmptyBetweenTimeRange(
+        final LocalDateTime queryStart,
+        final LocalDateTime queryEnd,
+        final String step
+    ) throws InterruptedException {
+
+        boolean valid = false;
+        while (!valid) {
+            try {
+                final Metrics serviceMetrics = queryMetrics(queryStart, queryEnd, step);
+
+                log.info("ServiceMetrics: {}", serviceMetrics);
+
+                AllOfMetricsMatcher instanceRespTimeMatcher = new AllOfMetricsMatcher();
+                MetricsValueMatcher equalsZero = new MetricsValueMatcher();
+                equalsZero.setValue("eq 0");
+                instanceRespTimeMatcher.setValue(equalsZero);
+                try {
+                    instanceRespTimeMatcher.verify(serviceMetrics);
+                    valid = true;
+                } catch (Throwable t) {
+                    log.warn("History metrics data are not deleted yet, {}", t.getMessage());
+                    Thread.sleep(10000);
+                }
+            } catch (Throwable t) {
+                log.warn("History metrics data are not deleted yet", t);
+                Thread.sleep(10000);
```

Comment:
I think it is more reasonable to set max test times, according to your `SW_CORE_DATA_KEEPER_EXECUTE_PERIOD` has been set to 1(min). Then, I would say 10 times(10 mins) should be good to say success or failure? What do you think?